lib/ddtrace/contrib/concurrent_ruby/context_composite_executor_service.rb

```diff
@@ -17,7 +17,7 @@ def initialize(composited_executor)
 
         # post method runs the task within composited executor - in a different thread
         def post(*args, &task)
-          tracer = Datadog.tracer
+          tracer = Datadog::Tracing.send(:tracer)
           parent_context = tracer.provider.context
 
           @composited_executor.post(*args) do
```

Comment:
I don't want integrations to have access to the context at all; this is a deliberate action to hide these internals that have been the source of so many breaking changes.

We do need to address the context management needs of some integrations, but I'd rather strip down the API surface area first, then re-add better context management APIs thoughtfully in 1.x.

Reply:
Fair enough; I'll leave the same suggestion as I did for resetting the context above: I think it'd be nice to create a private `provider_context` method in `Datadog::Tracing` so this can become `Datadog::Tracing.send(:provider_context)` and not have to reach inside the internals.

We could even call it `Datadog::Tracing.send(:deprecated_provider_context)` if we want to make it even more explicit that "here be dragons".

Reply:
See above.